src/NHibernate.Test/NHSpecificTest/GH3290/Fixture.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System.Collections.Generic;
+using NHibernate.Cfg;
+using NHibernate.Cfg.MappingSchema;
+using NHibernate.Mapping.ByCode;
+using NHibernate.Transform;
+using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.GH3290
+{
+	[TestFixture(true)]
+	[TestFixture(false)]
+	public class Fixture : TestCaseMappingByCode
+	{
+		private readonly bool _detectFetchLoops;
+
+		public Fixture(bool detectFetchLoops)
+		{
+			_detectFetchLoops = detectFetchLoops;
+		}
+
+		protected override HbmMapping GetMappings()
+		{
+			var mapper = new ModelMapper();
+			mapper.Class<Entity>(rc =>
+			{
+				rc.Id(x => x.Id, map => map.Generator(Generators.GuidComb));
+
+				rc.Property(
+					x => x.Name
+				);
+
+				rc.Set(
+					x => x.Children,
+					v =>
+					{
+						v.Table("EntityToEntity");
+						v.Cascade(Mapping.ByCode.Cascade.None);
+						v.Inverse(true);
+						v.Key(x =>
+						{
+							x.Column("ParentId");
+							x.NotNullable(true);
+						});
+						v.Lazy(CollectionLazy.Lazy);
+						v.Fetch(CollectionFetchMode.Join);
+					},
+					h => h.ManyToMany(m => m.Column("ChildId"))
+				);
+
+				rc.Set(
+					x => x.Parents,
+					v =>
+					{
+						v.Table("EntityToEntity");
+						v.Cascade(Mapping.ByCode.Cascade.All);
+
+						v.Key(x =>
+						{
+							x.Column("ChildId");
+							x.NotNullable(true);
+						});
+						v.Lazy(CollectionLazy.Lazy);
+						v.Fetch(CollectionFetchMode.Join);
+					},
+					h => h.ManyToMany(m => m.Column("ParentId"))
+				);
+			});
+
+			return mapper.CompileMappingForAllExplicitlyAddedEntities();
+		}
+
+		protected override void Configure(Configuration configuration)
+		{
+			if (_detectFetchLoops)
```

Comment:
Shouldn't we explicitly set `Environment.DetectFetchLoops` to false in this case so the test is not dependent on the default configuration (that can be anything).

Reply:
OK. Though it's quite specific setting